### PR TITLE
feat(metrics): add internal latency of actors, MVs and sinks

### LINKED ISSUE #18114: feat: metrics for each MV/Sink's latency
### Is your feature request related to a problem? Please describe.

Now we only have a global barrier latency. 

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR resolves #18114, and also introduced an actor-level metrics for #19619.

### New panel in **DEV** dashabord:

image

image

### New panel in **USER** dashabord:

image

Note there are 2 metrics for each sink - "output" means reading from log store, while "enqueue" means writing into log store.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by lmatz — APPROVED** [review]

File: grafana/risingwave-user-dashboard.dashboard.py [hunk 1]
```diff
@@ -744,6 +744,26 @@ def section_streaming(outer_panels):
                         ),
                     ],
                 ),
+                panels.timeseries_latency(
+                    "Latency of Materialize Views & Sinks",
```
Comment (xxchan) [inline grafana/risingwave-user-dashboard.dashboard.py:748]: This seems to be easier to interpret than the epoch one. Should we add it to dev dashboard also? (Actually I've never checked user dashboard before... 🤪)

**Review by xxchan — APPROVED** [review]
Looks great! Should we cherry-pick?

File: grafana/risingwave-dev-dashboard.dashboard.py [hunk 2]
```diff
@@ -1130,6 +1130,18 @@ def section_streaming(outer_panels):
                         ),
                     ],
                 ),
+                panels.timeseries_epoch(
+                    "Current Epoch of Materialize Views",
+                    "The current epoch that the Materialize Executors are processing. If an MV's epoch is far behind the others, "
+                    "it's very likely to be the performance bottleneck",
```
Comment (xxchan) [inline grafana/risingwave-dev-dashboard.dashboard.py:1136]: BTW, if an upstream MV is blocked, then all downstream MVs will have epoch lag? (Therefore we can't locate the root cause from this metrics alone) 🤔

Comment (xxchan) [reply grafana/risingwave-dev-dashboard.dashboard.py:1136]: I found you once raised the idea to show the epoch on DAG. 🤔

https://github.com/risingwavelabs/risingwave/issues/13481#issuecomment-2301341892

**Comment by fuyufjh** [comment]
> Looks great! Should we cherry-pick?

Do you mean check-pick to 2.1? Normally it will be shipped at 2.2, which sounds good to me

**Comment by xxchan** [comment]
Yes to 2.1. Since this may help troubleshooting incl. oncall, I feel it may be good to make it available eaelier...

**Pull request merged**